_posts/blog/2017-07-18-dwh-part2-we-need-a-storage.md

```diff
@@ -0,0 +1,154 @@
+* steep learning curve in order to boost performance
+* maybe training necessary (or even consultancy)
+* no native support for semi-structured data
+* database administration is complex (column compression, distribution, vacuumization, …)
+
+## The Opponent - Snowflake
+
+Redshift is not the only data warehouse as a service candidate. [Snowflake](https://www.snowflake.net/) is another one. 
+To be honest, we evaluated Snowflake a while ago. 
+A lot has changed since then, but let’s stick to the data we collected sometime in 2016.
+
+On Snowflake's homepage you can read:
+
+_From zero to Petabytes, our elastic data warehouse architecture scales to any volume of data painlessly and inexpensively. 
+No more painful decisions about what data you can afford to keep._
+
+__Pros:__
+
+* storage is cheap (maybe) -> see cons
+* native support for semi-structured data
+* (unlimited) scalability
+* good integration into AWS infrastructure (S3, AWS Lambdas, etc.)
+* database administration is done automatically
+
+__Cons:__
+
+* nontransparent pricing policy
+* missing information about tooling (ETL)
+
+Snowflake created a database layer on top of S3 (a little bit like Amazon Athena did later). Unfortunately it wasn’t easy to thoroughly test Snowflake at that time. 
+Consequently we did not investigate further. Maybe we would handle this issue differently today.
+
+## Finally
+
+**We decided for Redshift in the end.** I have to admit it hasn't been a too unbiased process, because more or less we wanted to try Redshift anyway. 
+
+**Starting with Redshift is easy, but getting satisfying results, especially if the amount of data increases, takes longer.** 
+Execution time and costs increase. You have to understand how Redshift works under the hood and you need to tune your tables accordingly. 
+Also its maintenance is not a piece of cake. We will share our findings in a later post.
+
+## Summary
+
+**Selecting a proper storage for a data warehouse is tough.** And it’s not at the same time. At the end of the day **you, as a data engineer, manage data, not a database**. 
+Data goes into the storage and is being retrieved again. Make sure both ways are comfortable. You, as a data engineer, are able to load data into the storage easily. 
+Your customer, usually some BI person, can fetch data quickly and is happy. Data maintenance should not be too complex as well. And all aggregations in between should also work smoothly. 
+
+__Questions you should ask:__
+
+* How does my infrastructure look like so far?
+* What are my sources?
+* How is my source data structured?
+* How much data do I expect to store?
+* Do I need unlimited scalability?
+* What tools (BI) need to access my data?
+* How much can I spend on it?
+* How complicated is the maintenance? 
+* Do we have capacity and knowledge to master this technology? 
+* Do we need training?
+
+
+And there's nothing wrong with having more than one storage. I mean, a storage is not Highlander in the end.
```

Comment:
"And there's nothing wrong with having more than one storage. I mean, a storage is not Highlander in the end." => "And there's nothing wrong with having more than one storage solution. I mean, storage is not Highlander in the end."